src/components/Components/AnimatedComponent/AnimatedComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, type ReactNode } from "react";
+import { useAnimation, motion, type Variants } from "framer-motion";
+import { useInView } from "react-intersection-observer";
+
+type AnimatedComponentProps = {
+  children: ReactNode;
+  animation?: Variants;
+  direction?: "left" | "right" | "top" | "bottom";
+};
+
+const AnimatedComponent = (props: AnimatedComponentProps) => {
```

Comment:
when you use children node, use PropsWithChildres<{typeName}> in type alias

<img width="706" alt="image" src="https://github.com/namu-labs-dev/victor1/assets/155515318/698862ed-016e-4a0a-a627-89b49a078528">
[Convention Docs[30-3]](https://www.notion.so/30-3-Component-with-children-Convention-d34b65ef326b48279adfcd8c6bf2cd98?pvs=4)

advantage
1. No Need to alias children type & schema
2. Strong Type check to children node